lifecycle-operator/webhooks/pod_mutator/handlers/objectmeta.go

```diff
@@ -94,24 +94,44 @@ func initEmptyAnnotations(meta *metav1.ObjectMeta, size int) {
 	}
 }
 
-func calculateVersion(pod *corev1.Pod) string {
+func getImageVersion(image string) (string, error) {
+	splitImage := strings.Split(image, ":")
+	lenImg := len(splitImage) - 1
+	if lenImg >= 1 && splitImage[lenImg] != "" && splitImage[lenImg] != "latest" {
+		return splitImage[lenImg], nil
+	}
+	return "", fmt.Errorf("Invalid image version")
+}
+
+func calculateVersion(pod *corev1.Pod, containerName string) (string, error) {
 	if len(pod.Spec.Containers) == 1 {
-		image := strings.Split(pod.Spec.Containers[0].Image, ":")
-		lenImg := len(image) - 1
-		if lenImg >= 1 && image[lenImg] != "" && image[lenImg] != "latest" {
-			return image[lenImg]
+		if containerName != "" && pod.Spec.Containers[0].Name != containerName {
+			return "", fmt.Errorf("The container name '%s' specified in %s does not match the name of the container in the pod", containerName, apicommon.ContainerNameAnnotation)
 		}
+		return getImageVersion(pod.Spec.Containers[0].Image)
 	}
 
 	name := ""
+	containerFound := false
 	for _, item := range pod.Spec.Containers {
+		if item.Name == containerName {
```

Comment:
would it be possible to cover this part of code with unit tests ?

Reply:
this gets tested [here](https://github.com/keptn/lifecycle-toolkit/blob/c70c29ada7fea2a7a4823aabf3454fa0a77625fc/lifecycle-operator/webhooks/pod_mutator/handlers/objectmeta_test.go#L243)